docs/design-documents/hal/0005-pin-names-Arduino-Uno-standard.md

```diff
@@ -0,0 +1,210 @@
+
+    // Arduino Uno (Rev3) connector pin connection naming  
+    // Px_xx relates to the processor pin connected to the Arduino Uno (Rev3) connector pin
+
+    ARDUINO_UNO_D0 = Px_xx,
+    ARDUINO_UNO_D1 = Px_xx,
+    ARDUINO_UNO_D2 = Px_xx,
+    ARDUINO_UNO_D3 = Px_xx,
+    ARDUINO_UNO_D4 = Px_xx,
+    ARDUINO_UNO_D5 = Px_xx,
+    ARDUINO_UNO_D6 = Px_xx,
+    ARDUINO_UNO_D7 = Px_xx,
+    ARDUINO_UNO_D8 = Px_xx,
+    ARDUINO_UNO_D9 = Px_xx,
+    ARDUINO_UNO_D10 = Px_xx,
+    ARDUINO_UNO_D11 = Px_xx,
+    ARDUINO_UNO_D12 = Px_xx,
+    ARDUINO_UNO_D13 = Px_xx,
+    ARDUINO_UNO_D14 = Px_xx,
+    ARDUINO_UNO_D15 = Px_xx,
+
+    ARDUINO_UNO_A0 = Px_xx,
+    ARDUINO_UNO_A1 = Px_xx,
+    ARDUINO_UNO_A2 = Px_xx,
+    ARDUINO_UNO_A3 = Px_xx,
+    ARDUINO_UNO_A4 = Px_xx,
+    ARDUINO_UNO_A5 = Px_xx,
+
+If the development board has the Arduino Uno connector in hardware, but does not comply with the Arduino Uno standard, whether it be with alternate functionality pins or no connected pins, the board should not be defined as Arduino Uno compliant and `ARDUINO_UNO` should not be added as a supported form factor in targets.json. Note this may result in a warning being generated at compile time to inform the user.
+
+The pins with alternate functions have to be referenced using MCU pin names (Px_xx) or pin name aliases in application code. 
+
+**I2C, SPI and UART definition**
+
+All I2C, SPI and UART pin name alias definitions for the Arduino Uno (Rev3) connector pins should be defined in ArduinoUnoAliases.h in the Mbed OS HAL (common to all Arduino Uno compliant targets) as follows:
+
+    #ifdef TARGET_FF_ARDUINO_UNO
+    // Arduino Uno I2C signals aliases
+    #define ARDUINO_UNO_I2C_SDA ARDUINO_UNO_D14
+    #define ARDUINO_UNO_I2C_SCL ARDUINO_UNO_D15
+
+    // Arduino Uno SPI signals aliases
+    #define ARDUINO_UNO_SPI_CS   ARDUINO_UNO_D10
+    #define ARDUINO_UNO_SPI_MOSI ARDUINO_UNO_D11
+    #define ARDUINO_UNO_SPI_MISO ARDUINO_UNO_D12
+    #define ARDUINO_UNO_SPI_SCK  ARDUINO_UNO_D13
+
+    // Arduino Uno UART signals aliases
+    #define ARDUINO_UNO_UART_TX  ARDUINO_UNO_D0
+    #define ARDUINO_UNO_UART_RX  ARDUINO_UNO_D1
+
+    #endif // TARGET_FF_ARDUINO_UNO
+
+**Other pin functions**
+
+Some Dx pinnames may support the usage of PWM or Timers functions. Although this is recomended, it's not a mandatory as requirement to be compliant with the Arduino Uno standard for Mbed boards.
+
+Note this might be one of the main differencess accross Mbed boards and therefore the application should not assume the same behaviour for PWM and Timers for them.
+
+The RESET signal should be defined in the Arduino Uno connector to let the user or an external component put the MCU in reset state.
+
+The VIN signal isn't strictly defined in the Arduino standard and in some cases may work as a bi-directional power supply. Tipically it could accept between 5 to 12 volts, but it's not a requirement.
```

Comment:
The Vin signal is strictly used in the Arduino Uno standard and should be implemented between 7V to 12V, a warning should be placed on someone's board if it isn't implemented in the correct voltage range.  if a user designs an Arduino Uno shield to expect 7V-12V on the Vin it will have power issues with a controller board supplying less then 7V and will likely cause the Arduino Uno shield to not power up correctly

Reply:
This will impact portability between development boards but this cannot be caught in software

Reply:
Added comments to the doc. We may need to elaborate on this and possible recommendations.